_includes/articles/wizardsardine-miniscript.md

```diff
@@ -0,0 +1,111 @@
+script has the same semantics?
+- In addition, Revault is using pre-signed transactions (to enforce spending
+policies). How can we know in advance the budget to allocate for fee-bumping
+given the configuration of the script? How can we make sure _any_ transaction
+spending those scripts will pass the most common standardness checks?
+- Finally, even assuming our scripts correspond to the intended semantics and
+are always spendable, how can we _concretely_ spend them? As in how can we
+produce a satisfying witness for ("sign for") each and every possible
+configuration? How can we make hardware signing devices compatible with our
+scripts?
+
+These questions would have been a show stopper if it was not for miniscript. Two
+guys in a garage are not going to write a software that [creates some script on
+the fly, hope for the best][rekt lost funds] and on top of that call it a
+security enhancing Bitcoin wallet. We wanted to start a company around the
+development of Revault, but we wouldn't get funding without providing some
+reasonable assurance to an investor we could bring a safe product to market. And
+we wouldn't be able to solve all these engineering problems without funding.
+
+[Enters miniscript][sipa miniscript], "a language for writing (a subset of)
+Bitcoin Scripts in a structured way, enabling analysis, composition, generic
+signing and more. [...] It has a structure that allows composition. It is very
+easy to statically analyze for various properties (spending conditions,
+correctness, security properties, malleability, ...)." This is exactly what we
+needed. Armed with this powerful tool, we could give better guarantees [0] to
+our investors, raise funds, and start the development of Revault.
+
+At the time, miniscript was still far from being a turnkey solution for any
+Bitcoin application developer to use. (If you are a new Bitcoin dev reading this
+after the year 2023, yes, there was a time where we used to write Bitcoin
+scripts BY HAND.) We had to integrate miniscript in Bitcoin Core (see PRs
+[#24147][Bitcoin Core #24147], [#24148][Bitcoin Core #24148] and
+[#24149][Bitcoin Core #24149]), which we used as the backend of the Revault
+wallet, and convince signing devices manufacturers to implement it in their
+firmware. The latter part would turn out to be the most difficult.
+
+It was a chicken-and-egg problem: incentives were low for manufacturers to
+implement miniscript with no demand from users. And we couldn't release Revault
+without signing device support. Fortunately this cycle was eventually broken by
+[Stepan Snigirev][] in March 2021 by [bringing][github embit descriptors]
+[support][github specter descriptors] for miniscript descriptors to the [Specter
+DIY][]. The Specter DIY was however for a long time disclaimed as being merely a
+"functional prototype", and [Salvatore Ingala][] brought [miniscript to a
+production-ready signing device] for the first time in 2022 with the [new
```

Comment:
Both links are missing. We can link to Salvatore's Twitter (https://twitter.com/salvatoshi) and to their announcement blog post (https://www.ledger.com/blog/miniscript-is-coming).

Reply:
Salvatore's link was in the footer for reference. Adding the ledger blog link.